meteor/utils.py

```diff
@@ -123,6 +123,7 @@ def average_phase_diff_in_degrees(
 
     diff = phase2 - phase1
     diff = (diff + 180) % 360 - 180
+    diff = diff[~np.isnan(diff)]
```

Comment:
Removing nans wholesale seems dangerous here. What if a numerical bug is introduced which causes some of `diff` to become nan. Can you trim array1 and array2 based on the intersections of their indices outside of this function? That would seem safer to me.

Reply:
Yep that is an alternative, and actually was the first approach I thought of. It adds a touch of complexity to the upstream code, but you are right, it's probably the most defensive (and therefore best) option.

An alternative would be to add an optional boolean argument `ignore_nans` to this function (`average_phase_diff_in_degrees`), but I think I like that one the least.

Let me try your suggestion in earnest and see how it fits.